scripts/install_moodle.sh

```diff
@@ -859,6 +859,11 @@ EOF
         sed -i "23 a \$CFG->alternative_file_system_class = '\\\tool_objectfs\\\azure_file_system';" /moodle/html/moodle/config.php
     fi
 
+   # setting file locking to false if fileServerType is set to nfs
+   if [ $fileServerType = "nfs" ]; then
+      sed -i "23 a \$CFG->preventfilelocking = false;" /moodle/html/moodle/config.php
```

Comment:
Thanks @superzoelu for your response with the `config-dist.php` comment, but then now I'm even more confused. If we have to disable file locking on NFS for that reason, shouldn't we set `$CFG->preventfilelocking` to `true`, not to `false`? I looked up Moodle code where `$CFG->preventfilelocking` is referenced (like https://github.com/moodle/moodle/blob/32f9550e8517c6238d0085f885e1a59add6eeb4d/lib/filestorage/file_system_filedir.php#L451-L455 ) and I see that setting the flag to false is just equivalent to not setting it at all. Are you sure you have correct understanding of this flag???

Besides, we never had issues with NFS without this switch (either false or true). The comment in the `config-dist.php` might be true for old versions of Linux for NFS, but I don't think it applies to current Linux kernels any more. See https://serverfault.com/questions/66919/file-locks-on-an-nfs --- "Since Linux 2.6.12, NFS clients support flock() locks …". I think this change is really not necessary for current Linux kernel versions.

I suppose we can still set this flag (but of course I believe it should be set to true, to make it effective), because according to the `config-dist.php` comment at https://github.com/moodle/moodle/blob/6d4bc5bd34809683e2e2ada3ce52a3e0adf94de1/config-dist.php#L525-L530 , the default locking for Postgres is DB locking, whereas the default locking for MySQL is file locking, implying that we can choose the DB locking no matter what by setting this flag to "true", but I doubt it will have any significant perf improvement. As far as I've heard, locking is mostly for cron jobs.